### DISCUSSION TITLE
Is there a way to print controls like PrintHelper in uwp?

### DISCUSSION THREAD

**Discussion by SongOfYouth** [unoplatform/uno#17404] [Q&A] [answered]
in uwp, there is a [PrintHelper](https://learn.microsoft.com/en-us/windows/communitytoolkit/helpers/printhelper) class for it. is it possibe to make it availadble in uno?

**Comment by jeromelaban** [comment]
Thanks for the question. Printing is a difficult topic (render content grabbing can vary greatly), and it's not been requested a lot, which is why we haven't looked at it yet. Still, you can use the native platform APIs to get you started.

Which platforms are of interest to you?

**Reply by SongOfYouth** [reply]
> Thanks for the question. Printing is a difficult topic (render content grabbing can vary greatly), and it's not been requested a lot, which is why we haven't looked at it yet. Still, you can use the native platform APIs to get you started.
> 
> Which platforms are of interest to you?

Thanks. I have tried it at uno 5.2 on windows like:
```csharp
var hWnd = WinRT.Interop.WindowNative.GetWindowHandle(App.CurrentWindow);
printManager = PrintManagerInterop.GetForWindow(hWnd);
printManager.PrintTaskRequested += PrintManager_PrintTaskRequested;
```
but it dosn't work(just show a print window and no content to print).

**Reply by jeromelaban** [reply]
> By a way,If i want to call it on wasm, i have to call it separately via js?

Yes, that's the way to do it, [using JSImport](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-interop.html?tabs=net7).

**Comment by SongOfYouth** [comment]
By a way,If i want to call it on wasm, i have to call it separately via js?

**Comment by SongOfYouth** [comment]
As the coming of version 6.0, maybe it's time to discuss this topic again? @jeromelaban

**Reply by jeromelaban** [reply, marked as answer]
We do not have a timeline for implementing this at this time. The introduction of Skia on all targets makes this topic a bit easier to implement, though.
